src/session/client-certificate.c

```diff
@@ -41,6 +43,19 @@
 
 #define CLIENT_CERTIFICATE_DIRECTORY   "/run/cockpit/tls/clients"
 
+static int
+open_proc_pid (pid_t pid)
+{
+    char path[100];
+    int r = snprintf (path, sizeof (path), "/proc/%d", pid);
+    if (r < 0 || r >= sizeof (path))
+        errx (EX, "memory error");
+    int fd = open (path, O_DIRECTORY | O_NOFOLLOW | O_PATH);
```

Comment:
Probably `O_CLOEXEC` is a good idea here.

Reply:
Fixed.